src/test/java/com/redhat/devtools/lsp4ij/features/formatting/TypeScriptClientSideFormatOnCompletionTriggerTest.java

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.formatting;
+
+import com.redhat.devtools.lsp4ij.fixtures.LSPClientSideOnTypeFormattingFixtureTestCase;
+
+/**
+ * TypeScript-based client-side on-type formatting tests for format-on-completion trigger.
+ */
+public class TypeScriptClientSideFormatOnCompletionTriggerTest extends LSPClientSideOnTypeFormattingFixtureTestCase {
```

Comment:
I've added three new test classes, one for each of the main aspects of this feature: format-on-close brace, format-on-statement terminator, and format-on-completion trigger.